ui-speedspacechart/src/components/helpers/layersManager.ts

```diff
@@ -1,16 +1,38 @@
 import * as d3 from 'd3';
 import type { Store } from '../../types/chartTypes';
 
-export const zoomX = (store: Store, setStore: React.Dispatch<React.SetStateAction<Store>>) =>
-  d3.zoom().on('zoom', (event) => {
-    const ratio = event.transform.k;
-    const leftOffset = event.transform.x;
-    setStore(() => ({
-      ...store,
-      ratio,
-      leftOffset,
-    }));
-  });
+export const zoom = (setStore: React.Dispatch<React.SetStateAction<Store>>) =>
+  d3
+    .zoom()
+    .filter((event) => {
+      return event.shiftKey;
+    })
+    .on('zoom', () => {
+      const canvas = d3.select('#front-interactivity-layer') as d3.Selection<
```

Comment:
Maybe store `#front-interactivity-layer` in a constant inside the const.ts file and reuse it wherever necessary

Reply:
done